app/src/main/java/com/taskmuse/app/ui/fragment/task/AddTaskFragment.java

```diff
@@ -0,0 +1,167 @@
+
+    public AddTaskFragment() {
+        // Required empty public constructor
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        return inflater.inflate(R.layout.fragment_add_task, container, false);
+    }
+
+    @Override
+    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+
+        // References to UI elements in the AddTaskFragment layout
+        Spinner projectNameSpinner = view.findViewById(R.id.projectNameSpinner);
+        Spinner prioritySpinner = view.findViewById(R.id.prioritySpinner);
+        Spinner statusSpinner = view.findViewById(R.id.statusSpinner);
+        Spinner taskAssigneeSpinner = view.findViewById(R.id.taskAssigneeSpinner);
+        TextInputEditText taskNameInput = view.findViewById(R.id.taskNameInput);
+        TextInputEditText descriptionInput = view.findViewById(R.id.descriptionInput);
+
+        // Populate spinners with data
+        List<String> projectNamesList = Arrays.asList("Project A", "Project B", "Project C", "Project D");
+        List<String> statusList = Arrays.asList("ToDo", "InProgress", "Done");
+        List<String> priorityList = Arrays.asList("High", "Highest", "Medium", "Low", "Lowest");
+        List<String> assigneeList = Arrays.asList("Kajal", "Agata", "PSM", "Sandesh");
+
+        ArrayAdapter<String> projectNameAdapter = new ArrayAdapter<>(requireContext(), android.R.layout.simple_spinner_item, projectNamesList);
+        projectNameAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
+        projectNameSpinner.setAdapter(projectNameAdapter);
+
+        ArrayAdapter<String> statusAdapter = new ArrayAdapter<>(requireContext(), android.R.layout.simple_spinner_item, statusList);
+        statusAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
+        statusSpinner.setAdapter(statusAdapter);
+
+        ArrayAdapter<String> priorityAdapter = new ArrayAdapter<>(requireContext(), android.R.layout.simple_spinner_item, priorityList);
+        priorityAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
+        prioritySpinner.setAdapter(priorityAdapter);
+
+        ArrayAdapter<String> assigneeAdapter = new ArrayAdapter<>(requireContext(), android.R.layout.simple_spinner_item, assigneeList);
+        assigneeAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
+        taskAssigneeSpinner.setAdapter(assigneeAdapter);
+
+        // Set OnClickListener for the "Add Task" button
+        Button addTaskButton = view.findViewById(R.id.addTaskButton);
+        addTaskButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                // Retrieve values from UI elements
+                String id = "";
```

Comment:
This is not required as it is creating a field which was not discussed as the format.

Reply:
I don't think we are using the ID anywhere specifically can we remove it from the Task.java file?

Reply:
It's being used in edit task fragment to assign ID from the collection